server/chat-plugins/teams.ts

```diff
@@ -156,19 +158,22 @@ export const TeamsHandler = new class {
 			}
 		}
 
-		const team = Teams.import(rawTeam, true);
-		if (!team) {
-			connection.popup('Invalid team:\n\n' + rawTeam);
+		const sets = Teams.import(team.packedTeam, true);
+		if (!sets) {
+			connection.popup('Invalid team:\n\n' + team.packedTeam);
 			return null;
 		}
-		if (team.length > 24) {
+		if (sets.length > 24) {
 			connection.popup("Your team has too many Pokemon.");
 		}
 		let unownWord = '';
-		// now, we purge invalid nicknames and make sure it's an actual team
-		// gotta use the validated team so that nicknames are removed
-		for (const set of team) {
-			set.name = this.isOMNickname(set.name, user) || set.species;
+		// make sure it's an actual team
+		for (const set of sets) {
+			if (!team.privacy) {
+				// if public, we purge invalid nicknames
```

Comment:
This doesn't work because people can toggle privacy without hitting `save`, meaning a user could upload malicious nicknames on a private team and then publicize it without hitting this gate.

Reply:
On reflection I think we should switch to removing nicknames on display instead of on save, which would let people sync nicknames even if they couldn't be displayed.

Reply:
Working code into place for this now. Will merge this PR after it's done.

Reply:
Sure. I'll handle the merge conflict.